treeherder/seta/job_priorities.py

```diff
@@ -0,0 +1,136 @@
+import datetime
+import logging
+
+from treeherder.model.models import Repository
+from treeherder.seta.common import unique_key
+from treeherder.seta.jobtypes import Treecodes
+from treeherder.seta.models import (JobPriority,
+                                    TaskRequest)
+
+LOG = logging.getLogger(__name__)
+RESET_DELTA = 5400
+
+
+class SetaError(Exception):
+    pass
+
+
+# XXX: We can get rid of this function if Treecodes took care of it
+def _map(build_system):
+    '''
+    Sample data from Treecodes().jobnames_query() (skipping irrelevant fields)
+    {
+        "buildplatform": "buildbot",
+        "buildtype": "debug",
+        "name": "web-platform-tests-1",
+        "platform": "windows8-64",
+        "ref_data_name": "Windows 8 64-bit mozilla-inbound debug test web-platform-tests-1",
+    },{
+        "buildplatform": "taskcluster",
+        "buildtype": "opt",
+        "name": "mochitest-webgl-e10s-1",
+        "platform": "linux64",
+        "ref_data_name": "desktop-test-linux64/opt-mochitest-webgl-e10s-1",
+    }
+    '''
+    map = {}
+    for job in Treecodes().jobnames_query():
+        if job['buildplatform'] == build_system:
+            key = unique_key(testtype=job['name'],
+                             buildtype=job['buildtype'],
+                             platform=job['platform'])
+            map[key] = job['ref_data_name']
+
+    return map
+
+
+def _process(build_system, job_priorities):
+    '''Return list of ref_data_name for job_priorities'''
+    jobs = []
+    # This map contains the ref_data_name of every Treeherder job
+    map = _map(build_system)
+
+    for jp in job_priorities:
+        if build_system == 'taskcluster':
+            # We need to retranslate the jobtype back to the proper data form after all.
+            # XXX: Ask jmaher, what would it take to fix this?
+            jp.testtype = jp.testtype.replace('e10s-browser-chrome', 'browser-chrome-e10s')
+            jp.testtype = jp.testtype.replace('e10s-devtools-chrome', 'devtools-chrome-e10s')
+            jp.testtype = jp.testtype.replace('gl-', 'webgl-')
```

Comment:
I don't think that bug will get fixed- so lets reference the bug in a comment here- possibly we can make a chain of Bugzilla bugs blocking the main bug so we have a list.